src/ctxmenu.ts

```diff
@@ -178,11 +187,21 @@ class ContextMenu implements CTXMenuSingleton {
     }
 
     private static addStylesToDom() {
+        // Only add styles once
+        if (this.stylesAdded) return;
+        
         if (document.readyState === "loading") {
             return document.addEventListener("readystatechange", this.addStylesToDom, { once: true });
         }
+        
+        this.stylesAdded = true;
+        
         //insert default styles as first css -> low priority -> user can overwrite it easily
         const style = document.createElement("style");
+        const nonce = ContextMenu.instance?.nonce;
+        if (nonce) {
+            style.nonce = nonce;
```

Comment:
empty string seems to be the default value.
Instead of using a local variable to get hold of `this.nonce`, I'd just do a single line:
`style.nonce = this.nonce ?? ""`
then you don't need an if and don't need the local `nonce` variable